libshaderc/src/shaderc_test.cc

```diff
@@ -216,6 +216,48 @@ TEST(CompileStringWithOptions, MacroCompileOptions) {
       shaderc_glsl_vertex_shader, cloned_options.get()));
 }
 
+TEST(CompileStringWithOptions, PreprocessingModeOnlyOption) {
+  Compiler compiler;
+  compile_options_ptr options(shaderc_compile_options_initialize());
+  shaderc_compile_options_set_preprocessing_only_mode(options.get());
+  ASSERT_NE(nullptr, compiler.get_compiler_handle());
+  const std::string kMinimalShader =
+      "#define E main\n"
+      "void E(){}\n";
+  Compilation comp(compiler.get_compiler_handle(), kMinimalShader,
+                   shaderc_glsl_vertex_shader, options.get());
+  EXPECT_TRUE(shaderc_module_get_success(comp.result()));
+  EXPECT_THAT(shaderc_module_get_bytes(comp.result()),
+              HasSubstr("void main(){ }"));
+
+  const std::string kMinimalShaderCloneOption =
+      "#define E_CLONE_OPTION main\n"
+      "void E_CLONE_OPTION(){}\n";
+  compile_options_ptr cloned_options(
+      shaderc_compile_options_clone(options.get()));
+  Compilation comp_clone(compiler.get_compiler_handle(),
+                         kMinimalShaderCloneOption, shaderc_glsl_vertex_shader,
+                         cloned_options.get());
+  EXPECT_TRUE(shaderc_module_get_success(comp_clone.result()));
+  EXPECT_THAT(shaderc_module_get_bytes(comp_clone.result()),
+              HasSubstr("void main(){ }"));
+}
+
+TEST(CompileStringWithOptions, DisassemblyModeOption){
+  Compiler compiler;
+  compile_options_ptr options(shaderc_compile_options_initialize());
+  shaderc_compile_options_set_disassembly_mode(options.get());
+  ASSERT_NE(nullptr, compiler.get_compiler_handle());
+  const std::string kMinimalShader = "void main(){}\n";
```

Comment:
Should also test that disassembly mode is carried into any clone of the options object.